spec/resources/user_resource_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe UserResource do
+  describe 'Serialization' do
+    it 'serializes attributes' do
+      user = create(:valid_user)
+      resource = UserResource.new(user, [])
+      serialized_json = resource.custom_json
+
+      expect(serialized_json[:data][:attributes][:email]).to eq(user.email)
+      expect(serialized_json[:data][:attributes][:name]).to eq(user.name)
+    end
+
+    it 'serializes relationships' do
+      user = create(:valid_user)
+      snippet = create(:valid_snippet, user: user)
+      resource = UserResource.new(user, [snippet])
+      serialized_json = resource.custom_json
+
+      expect(serialized_json[:data][:relationships][:snippets]).not_to be_empty
+    end
+
+    it 'serializes links' do
+      user = create(:valid_user)
+      resource = UserResource.new(user, [])
```

Comment:
Why pass an empty array as a param here?  You should also be using keyword argumenmts, which can have a default value in the method call.